kong/plugins/cors/handler.lua

```diff
@@ -25,21 +25,30 @@ local function configure_origin(ngx, conf)
   if #conf.origins == 1 then
     if conf.origins[1] == "*" then
       ngx.ctx.cors_allow_all = true
+      ngx.header["Access-Control-Allow-Origin"] = "*"
+      return
+    end
+
+    ngx.header["Vary"] = "Origin"
 
-    else
-      ngx.header["Vary"] = "Origin"
+    -- if this doesnt look like a regex, set the ACAO header directly
+    -- otherwise, we'll fall through to an iterative search and
+    -- set the ACAO header based on the client Origin
+    local from, to, err = re_find(conf.origins[1], "^[A-Za-z0-9.:/-]+$", "jo")
```

Comment:
Yes, I ran the linter before submitting as I think this came up once before? I'll make a note to look over linter behavior shortly.